manifests/config/resource.pp

```diff
@@ -64,7 +64,7 @@
   String                                      $resource_name   = $title,
   Enum['db', 'ldap']                          $type            = undef,
   String                                      $host            = undef,
-  Integer[1,65535]                            $port            = undef,
+  Optional[Integer[1,65535]]                  $port            = undef,
```

Comment:
If you don't set a port in your resource declaration, Puppet will throw an error: `Error while evaluating a Resource Statement, Icingaweb2::Config::Resource[Icingaweb2 Database]: parameter 'port' expects an Integer value, got Undef`.